### PR TITLE
Batch JSON-RPC requests

### PR DESCRIPTION
**Pull request by sz-piotr**

### CODE REVIEW

File: packages/buidler-core/src/internal/buidler-evm/jsonrpc/batch.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      const responses = await this._httpService.send(requests);
+      for (let i = 0; i < responses.length; i++) {
+        if (responses[i] instanceof Error) {
+          rejects[i](responses[i]);
+        } else {
+          resolves[i](responses[i]);
+        }
+      }
+    } catch (e) {
+      for (const reject of rejects) {
```
Comment (alcuadrado) [inline packages/buidler-core/src/internal/buidler-evm/jsonrpc/batch.ts:58]: This is a bit dangerous, as you can easily end up in a situation where a promise's `reject` is called twice, or its `reject` and `resolve` are called.

File: packages/buidler-core/src/internal/buidler-evm/jsonrpc/handler.ts [hunk 2]
```diff
@@ -106,7 +106,9 @@ export default class JsonRpcHandler {
           rpcReq.method === "eth_subscribe" &&
           isSuccessfulJsonResponse(rpcResp)
         ) {
-          subscriptions.push(rpcResp.result.id);
+          try {
+            subscriptions.push((rpcResp.result as any).id);
+          } catch {}
```
Comment (alcuadrado) [inline packages/buidler-core/src/internal/buidler-evm/jsonrpc/handler.ts:111]: Is this accidental?

Comment (sz-piotr) [reply packages/buidler-core/src/internal/buidler-evm/jsonrpc/handler.ts:111]: This is a result of proper types being applied to the jsonRpc responses

File: packages/buidler-core/src/internal/buidler-evm/jsonrpc/batch.ts (hunk 1 above)
Comment (sz-piotr) [reply packages/buidler-core/src/internal/buidler-evm/jsonrpc/batch.ts:58]: You are right. I'll investigate a better way to handle this

**Comment by alcuadrado** [comment]
After playing with this implementation for some time, I decided not to merge it. The reason is that most of the cases it fails to batch requests that could be batched.

I replaced this PR with #806, which is more explicit, and simpler.

**Pull request closed without merging**